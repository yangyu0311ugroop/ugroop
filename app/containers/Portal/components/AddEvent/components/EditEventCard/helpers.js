import dotProp from 'dot-prop-immutable';
import get from 'lodash/get';
import { EVENT_VIEW_HELPERS } from 'smartComponents/Node/types/Event/helpers';

export const convertReservation = data => {
  if (!EVENT_VIEW_HELPERS.isTransportation(data)) {
    const eventBooking = get(data, `detail.detail.common.eventBooking`);

    if (!eventBooking) return data;

    return dotProp.merge(data, 'detail', {
      bookingNumber: eventBooking.bookingNumber,
      bookedBy: eventBooking.bookedBy,
      supplier: eventBooking.supplierName,
      supplierPhoneNumber: eventBooking.supplierNumber,
      bookingPersonCount: eventBooking.personCount,
    });
  }

  const detail = get(data, `detail`);

  return dotProp.merge(data, 'detail.detail.common.eventBooking', {
    bookingNumber: detail.bookingNumber,
    bookedBy: detail.bookedBy,
    supplierName: detail.supplier,
    supplierNumber: detail.supplierPhoneNumber,
    personCount: detail.bookingPersonCount,
  });
};

export const EDIT_HELPERS = {
  convertReservation,
};
