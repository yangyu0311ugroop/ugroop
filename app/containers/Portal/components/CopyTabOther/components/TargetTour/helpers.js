const createModel = data => {
  const {
    firstName,
    lastName,
    email,
    dob,
    dobMode,
    namePersonId,
    personType,
    comment,
    note,
    primary,
    phone,
    phoneId,
    ageType,
    honorificTitle,
    orgId,
  } = data;
  return {
    firstName,
    lastName,
    email,
    dob,
    dobMode,
    namePersonId,
    personType,
    comment,
    note,
    primary,
    phone,
    phoneId,
    ageType,
    honorificTitle,
    orgId,
  };
};

const getParticipant = (nodes, { firstName, lastName, email }) =>
  nodes
    .map(node => node.customData)
    .filter(
      person =>
        (person.firstName === firstName && person.lastName === lastName) ||
        (person.email === email && !!person.email),
    );

export const MODEL_HELPERS = {
  createModel,
  getParticipant,
};
