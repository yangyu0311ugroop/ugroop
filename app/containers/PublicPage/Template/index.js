import {
  GET_PUB_TEMPLATE_PEOPLE,
  PUB_API,
  GET_PUB_TEMPLATE_HEADER,
  GET_PUB_TEMPLATE_EVENTS,
} from 'apis/constants';
import React, { PureComponent } from 'react';
import 'url-search-params-polyfill';
import PropTypes from 'prop-types';
import GridItem from 'components/GridItem';
import GridContainer from 'components/GridContainer';
import { get } from 'lodash';
import resaga, { reducer } from 'resaga';
import { compose } from 'redux';
import { REDIRECT } from 'utils/pageRedirect';
import { withStyles } from '@material-ui/core/styles';
import injectReducer from 'utils/injectReducer';
import VisibleChildren from 'smartComponents/Node/logics/VisibleChildren';
import {
  CONFIG,
  TEMPLATE_CONFIG,
  PUBLIC_TEMPLATE,
  FETCH_TEMPLATE,
  TAB_GALLERY,
} from './config';
import PublicTabPage from './components/PubTemplateTabs';
import PubTemplateHeader from './components/PubTemplateHeader';
import styles from './styles';

export class PublicTemplatePageIndex extends PureComponent {
  componentDidMount = () => this.fetchData();

  componentWillReceiveProps = nextProps => {
    if (nextProps.currentQueryDayId !== this.props.currentQueryDayId) {
      const { location, history, timelineIndex } = this.props;
      const url = `${location.pathname}?tab=${timelineIndex}&dayId=${
        nextProps.currentQueryDayId
      }`;
      history.push(url);
    }
  };

  fetchData = () => {
    const hashkey = get(this, 'props.match.params.hashkey', 0);
    if (hashkey) {
      this.props.resaga.dispatchTo(PUB_API, GET_PUB_TEMPLATE_HEADER, {
        payload: { hashkey },
        onError: this.fetchError,
      });
      this.props.resaga.dispatchTo(PUB_API, GET_PUB_TEMPLATE_PEOPLE, {
        payload: { hashkey },
      });
      this.props.resaga.dispatchTo(PUB_API, GET_PUB_TEMPLATE_EVENTS, {
        payload: { hashkey },
      });
    }
  };

  fetchError = () => {
    const browserHistory = this.props.history;
    browserHistory.push(REDIRECT.STATUS_RESOURCE_NOT_FOUND);
  };

  render() {
    const { classes, createdBy, templateId, location } = this.props;
    const isLoading = this.props.resaga.isLoading(FETCH_TEMPLATE);
    const hashkey = get(this, 'props.match.params.hashkey', 0);
    const { search } = location;
    const params = new URLSearchParams(search);
    const activeTab = parseInt(params.get('tab'), 10) || 0;
    let currentDayId = -1;
    if (activeTab === 0) {
      currentDayId = parseInt(params.get('dayId'), 10) || -1;
    }
    let content = 'Loading';
    if (!templateId) return content;
    if (!isLoading) {
      content = (
        <PublicTabPage
          hashkey={hashkey}
          templateId={templateId}
          activeTabIndex={activeTab}
          currentQueryDayId={currentDayId}
        />
      );
    }
    return (
      <React.Fragment>
        <VisibleChildren id={templateId} />

        <GridContainer spacing={0} direction="column">
          <GridItem xs>
            <PubTemplateHeader templateId={templateId} createdBy={createdBy} />
          </GridItem>
          <GridItem className={classes.content}>{content}</GridItem>
        </GridContainer>
      </React.Fragment>
    );
  }
}

PublicTemplatePageIndex.propTypes = {
  classes: PropTypes.object.isRequired,
  history: PropTypes.object,
  resaga: PropTypes.object,
  location: PropTypes.object,
  // resaga
  createdBy: PropTypes.number,
  templateId: PropTypes.number,
  currentQueryDayId: PropTypes.number,
  timelineIndex: PropTypes.number,
};

PublicTemplatePageIndex.defaultProps = {
  location: { pathname: '' },
  templateId: 0,
  timelineIndex: 0,
};

const templateManagementReducer = injectReducer({
  key: PUBLIC_TEMPLATE,
  reducer: reducer(PUBLIC_TEMPLATE),
});
const tabGalleryReducer = injectReducer({
  key: TAB_GALLERY,
  reducer: reducer(TAB_GALLERY),
});
export default compose(
  templateManagementReducer,
  tabGalleryReducer,
  withStyles(styles, { name: 'PublicTemplate' }),
  resaga(TEMPLATE_CONFIG),
  resaga(CONFIG),
)(PublicTemplatePageIndex);
