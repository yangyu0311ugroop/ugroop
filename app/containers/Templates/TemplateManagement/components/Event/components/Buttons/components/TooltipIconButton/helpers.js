import { EVENT_CONSTANTS } from 'utils/constants/events';

const showSubIconFn = (type, subtype, props = {}) => {
  const { showSubIcon } = props;

  if (!showSubIcon) return false;

  const exception = [EVENT_CONSTANTS.TRANSPORTATIONS.FLIGHT.type];

  return exception.indexOf(subtype) === -1;
};

export const EVENT_ICON_HELPERS = {
  showSubIconFn,
};
