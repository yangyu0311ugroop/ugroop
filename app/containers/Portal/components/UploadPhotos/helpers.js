import { DEFAULT, UPLOADED } from 'appConstants';
import dotProp from 'dot-prop-immutable';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { ACTIVITY } from 'utils/modelConstants';

const gridSize = count =>
  LOGIC_HELPERS.switchCase(count, {
    1: { xs: 12 },
    2: { xs: 12, sm: 6 },
    [DEFAULT]: { xs: 12, sm: 6, md: 4, lg: 3 },
  });

const maxWidth = count => LOGIC_HELPERS.ifElse(count <= 2, 'sm', 'md');

const rotateLeftFn = rotate => {
  if (!rotate || typeof rotate !== 'number') return -90;
  if (rotate === -270) return 0;

  return rotate - 90;
};

const rotateRightFn = rotate => {
  if (!rotate || typeof rotate !== 'number') return 90;
  if (rotate === 270) return 0;

  return rotate + 90;
};

const normaliseSubmitData = ({ id, form, uploadedFiles, droppedFiles }) => (
  accu,
  fileId,
) => {
  const file = dotProp.get(droppedFiles, `${fileId}`);

  if (!file) return accu;

  const { status, rotate, errorMessage } = file;
  const photo = dotProp.get(uploadedFiles, `${fileId}.url`);
  const description = dotProp.get(form, `${fileId}.description`);

  if (status !== UPLOADED || errorMessage) {
    return accu;
  }

  return accu.concat({
    type: ACTIVITY,
    parentNodeId: id,
    customData: {
      description,
      photo,
      metaInfo: {
        rotate,
      },
    },
  });
};

export const UPLOAD_PHOTO_UTILS = {
  gridSize,
  maxWidth,
  rotateLeftFn,
  rotateRightFn,
  normaliseSubmitData,
};
