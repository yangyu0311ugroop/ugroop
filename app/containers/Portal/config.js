import { PORTAL_HELPERS, PORTAL_IDS_SELECTOR } from 'containers/Portal/helpers';

export const CONFIG = {
  value: {
    portalIds: PORTAL_IDS_SELECTOR,
  },
  setValue: {
    ...PORTAL_HELPERS.setValue,
  },
};
