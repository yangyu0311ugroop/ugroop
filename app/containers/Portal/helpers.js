import {
  ADD_ROUTE,
  PHOTO_PREVIEW,
  MANAGE_TABS,
  PROMPT,
  UPLOAD_PHOTOS,
  USER_DATA_STORE,
  ADD_RISK,
  VIEW_RISK,
  ADD_HAZARD,
  CREATE_CHAT_CHANNEL,
  EDIT_CHAT_DESCRIPTION,
  ADD_PEOPLE_IN_CHANNEL,
  DATES_SELECT,
  POPPER_DIALOG,
  HIDE_CHAT_DESCRIPTION,
  LEAVE_CHAT_DESCRIPTION,
  DELETE_CHAT_DESCRIPTION,
  CLONE_TEMPLATE,
  VIEW_EVENT,
  ADD_ROOM,
  VIEW_ROOM,
  ADD_ROOM_PAX,
  CREATE_TOUR_EMAIL,
  ADD_EVENT,
  PRINT_PDF_FORM,
  ADD_TOUR,
  COPY_PARTICIPANT,
  PARTICIPANT_LIST,
  ADD_TAB,
  COPY_TAB_OTHER,
  SHOW_SYSTEM_UPDATE,
  HELP_DIALOG,
  ADD_EDIT_INSURANCE,
} from 'appConstants';
import { DATA_HELPERS } from 'datastore/utils';
import React from 'react';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { STRING_HELPERS } from 'utils/stringAdditions';

const openPortal = type => (
  portalProps,
  { resaga },
  portalId,
  now = Date.now(),
) => {
  const id = portalId || now;
  resaga.setValue({
    portalIds: DATA_HELPERS.arrayAdd(id),
    portalData: DATA_HELPERS.objectAdd({
      [id]: { type, ...portalProps },
    }),
  });

  return id;
};

const closePortal = (id, { resaga }) => {
  resaga.setValue({
    portalIds: DATA_HELPERS.arrayRemove(id),
    portalData: DATA_HELPERS.objectRemove(id),
  });
};

// convenience functions, optional, can call openPortal(TYPE)(...props) instead
const openAddRoute = openPortal(ADD_ROUTE);
const openAddRisk = openPortal(ADD_RISK);
const openAddHazard = openPortal(ADD_HAZARD);
const openAddRoom = openPortal(ADD_ROOM);
const openAddRoomPax = openPortal(ADD_ROOM_PAX);
const openCreateChatChannel = openPortal(CREATE_CHAT_CHANNEL);
const openEditChatChannel = openPortal(EDIT_CHAT_DESCRIPTION);
const openAddPeopleInChannel = openPortal(ADD_PEOPLE_IN_CHANNEL);
const openArchiveChannelDialog = openPortal(HIDE_CHAT_DESCRIPTION);
const openLeaveChannelDialog = openPortal(LEAVE_CHAT_DESCRIPTION);
const openDeleteChannelDialog = openPortal(DELETE_CHAT_DESCRIPTION);
const openViewEvent = (portalProps, props, portalId, now) =>
  openPortal(VIEW_EVENT)(
    { flightBookingId: null, id: null, selectedId: null, ...portalProps },
    props,
    portalId,
    now,
  );
const openAddTab = openPortal(ADD_TAB);
const openAddEvent = openPortal(ADD_EVENT);
const openAddTour = openPortal(ADD_TOUR);
const openDatesSelect = openPortal(DATES_SELECT);
const openPopperDialog = openPortal(POPPER_DIALOG);
const openViewRisk = openPortal(VIEW_RISK);
const openViewRoom = openPortal(VIEW_ROOM);
const openPhotoPreview = openPortal(PHOTO_PREVIEW);
const openUploadPhotos = openPortal(UPLOAD_PHOTOS);
const openManageTabs = openPortal(MANAGE_TABS);
const prompt = openPortal(PROMPT);
const openCloneMarketTemplate = openPortal(CLONE_TEMPLATE);
const openCreateTourEmail = openPortal(CREATE_TOUR_EMAIL);
const DANGER = 'delete';
const openPrintPdfForm = openPortal(PRINT_PDF_FORM);
const openCopyParticipant = openPortal(COPY_PARTICIPANT);
const openParticipantList = openPortal(PARTICIPANT_LIST);
const openCopyTabOther = openPortal(COPY_TAB_OTHER);
const openHelpDialog = openPortal(HELP_DIALOG);
const showSystemUpdate = openPortal(SHOW_SYSTEM_UPDATE);
const showAddEditInsurance = openPortal(ADD_EDIT_INSURANCE);

const confirmDeletePhoto = ({ count, onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: (
        <>
          Are you sure you want to delete{' '}
          {LOGIC_HELPERS.ifElse(
            count > 1,
            `the selected ${STRING_HELPERS.pluralise('image', count)}`,
            'this image',
          )}
          ?
        </>
      ),
      dialogTitle: `Delete ${STRING_HELPERS.pluralise('Image', count)}`,
      confirmButton: LOGIC_HELPERS.ifElse(
        count > 1,
        `Delete ${count} Selected ${STRING_HELPERS.pluralise('Image', count)}`,
        'Delete',
      ),
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmCancelUploadPhotos = ({ onConfirm }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      onConfirm,
      simplifyDialog: true,
      headlineText: "If you discard now, you'll lose your progress.",
      dialogTitle: 'Discard Changes?',
      confirmButton: 'Discard',
      cancelButton: 'Go Back',
    },
    props,
  );
};

const promptEventAdded = ({ name, type = 'Event' }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: 'confirm',
      confirmation: true,
      simplifyDialog: true,
      headlineText: `${LOGIC_HELPERS.ifElse(
        name,
        `"${name}"`,
        type,
      )} has been added successfully.`,
      dialogTitle: 'Success',
      confirmButton: 'OK',
    },
    props,
  );
};

const confirmCancelAddRoute = ({ onConfirm }, props) => {
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      onConfirm,
      simplifyDialog: true,
      headlineText: "If you discard now, you'll lose your progress.",
      dialogTitle: 'Success',
      confirmButton: 'OK',
    },
    props,
  );
};

const confirmDeleteRoute = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to delete this route?',
      dialogTitle: `Delete Route`,
      confirmButton: 'Delete',
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmDeleteBooking = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to delete this flight itinerary?',
      dialogTitle: `Delete flight itinerary`,
      confirmButton: 'Delete',
      cancelButton: 'Go Back',
    },
    props,
  );

const confirmDelete = (
  {
    onConfirm,
    message,
    title,
    closeOnConfirm,
    confirmButton = 'Delete',
    cancelButton = 'Go Back',
  },
  props,
) =>
  PORTAL_HELPERS.prompt(
    {
      simplifyDialog: true,
      template: DANGER,
      closeOnConfirm,
      onConfirm,
      headlineText: message,
      dialogTitle: title,
      confirmButton,
      cancelButton,
    },
    props,
  );

const confirmCustom = (
  {
    onConfirm,
    message,
    title,
    closeOnConfirm,
    confirmButton = 'Continue',
    cancelButton = 'Go Back',
    template = 'confirm',
  },
  props,
) =>
  PORTAL_HELPERS.prompt(
    {
      simplifyDialog: true,
      template,
      closeOnConfirm,
      onConfirm,
      headlineText: message,
      dialogTitle: title,
      confirmButton,
      cancelButton,
    },
    props,
  );

const confirmCancelTransfer = ({ onConfirm }, props) =>
  PORTAL_HELPERS.prompt(
    {
      template: DANGER,
      closeOnConfirm: false,
      onConfirm,
      simplifyDialog: true,
      headlineText: 'Are you sure you want to cancel pending transfer?',
      dialogTitle: `Cancel Transfer of Ownership`,
      confirmButton: 'Continue',
      cancelButton: 'Go Back',
    },
    props,
  );
export const PORTAL_IDS_SELECTOR = [USER_DATA_STORE, 'portal', 'portalIds'];
export const PORTAL_DATA_SELECTOR = [USER_DATA_STORE, 'portal', 'portalData'];
export const PORTAL_DATA_ID_SELECTOR = ({ portalId }) => [
  USER_DATA_STORE,
  'portal',
  'portalData',
  portalId,
];
const setValue = {
  portalIds: PORTAL_IDS_SELECTOR,
  portalData: PORTAL_DATA_SELECTOR,
};

const close = ({ onClose, portalId }) => {
  LOGIC_HELPERS.ifFunction(onClose, [portalId]);
};

export const PORTAL_HELPERS = {
  openPhotoPreview,
  openAddRoute,
  openAddRisk,
  openAddHazard,
  openAddRoom,
  openAddRoomPax,
  openViewEvent,
  openAddTab,
  openAddEvent,
  openAddTour,
  openDatesSelect,
  openViewRisk,
  openViewRoom,
  openUploadPhotos,
  openManageTabs,
  openCreateChatChannel,
  openEditChatChannel,
  openAddPeopleInChannel,
  openArchiveChannelDialog,
  openLeaveChannelDialog,
  openDeleteChannelDialog,
  openCloneMarketTemplate,
  openCreateTourEmail,
  openPortal,
  closePortal,
  close,
  prompt,
  confirmDeletePhoto,
  confirmDeleteRoute,
  confirmDelete,
  confirmCancelUploadPhotos,
  promptEventAdded,
  confirmCancelAddRoute,
  setValue,
  confirmCancelTransfer,
  openPopperDialog,
  openPrintPdfForm,
  openCopyParticipant,
  confirmCustom,
  openParticipantList,
  openCopyTabOther,
  confirmDeleteBooking,
  openHelpDialog,
  showSystemUpdate,
  showAddEditInsurance,
};
