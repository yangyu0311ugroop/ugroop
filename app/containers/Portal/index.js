import { DATA_HELPERS } from 'datastore/utils';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { compose } from 'redux';
import resaga from 'resaga';
import Dialog from './components/Dialog';
import { CONFIG } from './config';

export class Portal extends PureComponent {
  closePortal = id => {
    this.props.resaga.setValue({
      portalIds: DATA_HELPERS.arrayRemove(id),
      portalData: DATA_HELPERS.objectRemove(id),
    });
  };

  render = () => {
    const { portalIds } = this.props;

    return portalIds.map(portalId => (
      <Dialog key={portalId} portalId={portalId} onClose={this.closePortal} />
    ));
  };
}

Portal.propTypes = {
  // hoc props
  resaga: PropTypes.object.isRequired,

  // parent props

  // resaga props
  portalIds: PropTypes.array,
};

Portal.defaultProps = {
  portalIds: [],
};

export default compose(resaga(CONFIG))(Portal);
