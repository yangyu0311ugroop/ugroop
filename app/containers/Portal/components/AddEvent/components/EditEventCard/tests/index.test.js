import { NODE_API_HELPERS } from 'apis/components/Node/helpers';
import { TEMPLATE_API_HELPERS } from 'apis/components/Template/helpers';
import { PORTAL_HELPERS } from 'containers/Portal/helpers';
import { shallow } from 'enzyme';
import React from 'react';
import { TEST_HELPERS } from 'utils/helpers/testHelpers';
import { EditEventCard } from '../index';

jest.useFakeTimers();

describe('<EditEventCard />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const props = {
    classes: {},
    resaga,
  };

  beforeEach(() => {
    jest.clearAllMocks();

    rendered = shallow(<EditEventCard {...props} />);
    instance = rendered.instance();
  });

  it('should exists', () => {
    expect(EditEventCard).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('patchEvent()', () => {
    it('should patchEvent() FlightBookingForm', () => {
      TEMPLATE_API_HELPERS.patchFlightBooking = jest.fn();
      rendered.setProps({ subform: 'FlightBookingForm' });

      instance.patchEvent({})();

      TEST_HELPERS.expectCalled(TEMPLATE_API_HELPERS.patchFlightBooking);
    });

    it('should patchEvent()', () => {
      TEMPLATE_API_HELPERS.patchEvent = jest.fn();

      instance.patchEvent({})();

      TEST_HELPERS.expectCalled(TEMPLATE_API_HELPERS.patchEvent);
    });
  });

  describe('handleValidSubmit()', () => {
    it('should close if !isChanged', () => {
      const onClose = jest.fn();
      rendered.setProps({ onClose });
      rendered.setState({ isChanged: false });

      instance.handleValidSubmit();

      TEST_HELPERS.expectCalled(onClose);
    });

    it('should return null if creating', () => {
      rendered.setState({ creating: true, isChanged: true });

      expect(instance.handleValidSubmit()).toBe(null);
    });

    it('should handleValidSubmit', () => {
      rendered.setState({ creating: false, isChanged: true });
      instance.patchEvent = jest.fn();

      instance.handleValidSubmit();

      expect(rendered.state().creating).toBe(true);
    });
  });

  describe('changeFormValue()', () => {
    it('should changeFormValue()', () => {
      instance.changeFormValue({ data: 'formValue' });

      expect(rendered.state().formValue).toEqual({ data: 'formValue' });
    });
  });

  describe('handlePatchEventSuccess()', () => {
    it('should handlePatchEventSuccess()', () => {
      instance.updateTimes = jest.fn();
      PORTAL_HELPERS.openViewEvent = jest.fn();

      instance.handlePatchEventSuccess();

      TEST_HELPERS.expectCalled(instance.updateTimes);
    });
  });

  describe('updateTimes()', () => {
    it('should updateTimes()', () => {
      NODE_API_HELPERS.getTimes = jest.fn();

      instance.updateTimes();

      TEST_HELPERS.expectCalled(NODE_API_HELPERS.getTimes);
    });
  });

  describe('formChangeDebounce()', () => {
    it('should call once', () => {
      instance.formChange = jest.fn();

      instance.formChangeDebounce({ placeId: 1 });
      instance.formChangeDebounce({ placeId: 1 });

      // Fast-forward until all timers have been executed
      jest.runAllTimers();

      expect(instance.formChange).toBeCalledTimes(1);
    });
  });

  describe('formChange()', () => {
    it('should formChange()', () => {
      instance.formChange({ 'data.name': 'aabbcc' });

      expect(rendered.state().formValue).toEqual({ data: { name: 'aabbcc' } });
    });
  });

  describe('handleValid()', () => {
    it('should handleValid()', () => {
      instance.handleValid(true)();

      expect(rendered.state().isValid).toBe(true);
    });
  });

  describe('renderForm()', () => {
    it('should renderForm', () => {
      TEST_HELPERS.expectMatchSnapshot(instance.renderForm);
    });
  });

  describe('renderLeft()', () => {
    it('should renderLeft smDown', () => {
      rendered.setProps({ smDown: true });
      instance.renderForm = jest.fn(() => 'renderForm');

      TEST_HELPERS.expectMatchSnapshot(instance.renderLeft);
    });

    it('should renderLeft', () => {
      rendered.setProps({ smDown: false });
      instance.renderForm = jest.fn(() => 'renderForm');

      TEST_HELPERS.expectMatchSnapshot(instance.renderLeft);
    });
  });

  describe('makeFormValue()', () => {
    it('should work when changing type', () => {
      rendered.setState({
        formValue: {
          data: {
            type: 'type',
            detail: {
              type: 'subtype',
            },
          },
          temp: {
            changingType: true,
          },
        },
      });

      expect(instance.makeFormValue()).toBeDefined();
    });

    it('should makeFormValue', () => {
      rendered.setState({
        formValue: {
          data: {
            type: 'type',
            detail: {
              type: 'subtype',
            },
          },
          temp: {
            changingType: true,
          },
        },
      });

      expect(instance.makeFormValue()).toBeDefined();
    });
  });

  describe('makeSubFormValue()', () => {
    it('should makeSubFormValue subform AttachmentForm', () => {
      rendered.setProps({ subform: 'AttachmentForm' });

      expect(instance.makeSubFormValue()).toBeDefined();
    });

    it('should makeSubFormValue subform ReservationForm', () => {
      rendered.setProps({ subform: 'ReservationForm' });

      expect(instance.makeSubFormValue()).toBeDefined();
    });

    it('should makeSubFormValue subform CancellationForm', () => {
      rendered.setProps({ subform: 'CancellationForm' });

      expect(instance.makeSubFormValue()).toBeDefined();
    });

    it('should makeSubFormValue normal', () => {
      rendered.setProps({ subform: '' });

      expect(instance.makeSubFormValue()).toBeDefined();
    });
  });

  describe('renderEvent()', () => {
    it('should renderEvent', () => {
      TEST_HELPERS.expectMatchSnapshot(instance.renderEvent);
    });
  });

  describe('renderContent()', () => {
    it('should renderContent smDown', () => {
      rendered.setProps({ smDown: true });
      instance.renderLeft = jest.fn(() => 'renderLeft');

      TEST_HELPERS.expectMatchSnapshot(instance.renderContent);
    });

    it('should renderContent', () => {
      rendered.setProps({ smDown: false });
      instance.renderLeft = jest.fn(() => 'renderLeft');
      instance.renderEvent = jest.fn(() => 'renderEvent');

      TEST_HELPERS.expectMatchSnapshot(instance.renderContent);
    });
  });

  describe('render()', () => {
    it('should render', () => {
      instance.renderContent = jest.fn(() => 'renderContent');

      TEST_HELPERS.expectMatchSnapshot(instance.render);
    });
  });
});
