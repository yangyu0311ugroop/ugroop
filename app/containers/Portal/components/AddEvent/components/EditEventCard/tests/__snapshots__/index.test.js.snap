// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditEventCard /> render() should render 1`] = `
<div>
  <WithStyles(GridContainer)
    spacing={0}
    wrap="nowrap"
  >
    <WithStyles(GridItem)>
      <div>
        <Formsy
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <WithStyles(Component)
            currentData={
              Object {
                "data": Object {},
                "node": Object {},
              }
            }
            disabled={true}
            formValue={
              Object {
                "data": Object {
                  "detail": Object {
                    "type": undefined,
                  },
                  "name": "",
                  "type": "Activity",
                },
                "node": Object {
                  "customData": Object {
                    "start": Object {
                      "tempDay": "",
                    },
                  },
                },
              }
            }
            onChangeFormValue={[Function]}
          />
        </Formsy>
      </div>
    </WithStyles(GridItem)>
    <WithStyles(GridItem)
      className=""
      xs={true}
    >
      <div>
        <WithStyles(GridContainer)
          alignItems="center"
          direction="column"
          spacing={0}
        >
          <WithStyles(GridItem)
            className=""
          >
            <WithStyles(ViewEvent)
              data={
                Object {
                  "detail": Object {
                    "type": undefined,
                  },
                  "name": "",
                  "type": "Activity",
                }
              }
              node={
                Object {
                  "customData": Object {
                    "start": Object {
                      "tempDay": "",
                    },
                  },
                }
              }
              styled={true}
            />
          </WithStyles(GridItem)>
        </WithStyles(GridContainer)>
      </div>
    </WithStyles(GridItem)>
  </WithStyles(GridContainer)>
</div>
`;

exports[`<EditEventCard /> renderContent() should renderContent 1`] = `<div />`;

exports[`<EditEventCard /> renderContent() should renderContent smDown 1`] = `<div />`;

exports[`<EditEventCard /> renderEvent() should renderEvent 1`] = `
<div>
  <WithStyles(GridContainer)
    alignItems="center"
    direction="column"
    spacing={0}
  >
    <WithStyles(GridItem)
      className=""
    >
      <WithStyles(ViewEvent)
        data={
          Object {
            "detail": Object {
              "type": undefined,
            },
            "name": "",
            "type": "Activity",
          }
        }
        node={
          Object {
            "customData": Object {
              "start": Object {
                "tempDay": "",
              },
            },
          }
        }
        styled={true}
      />
    </WithStyles(GridItem)>
  </WithStyles(GridContainer)>
</div>
`;

exports[`<EditEventCard /> renderForm() should renderForm 1`] = `
<div>
  <Formsy
    disabled={false}
    getErrorMessage={[Function]}
    getErrorMessages={[Function]}
    getValue={[Function]}
    hasValue={[Function]}
    isFormDisabled={[Function]}
    isFormSubmitted={[Function]}
    isPristine={[Function]}
    isRequired={[Function]}
    isValid={[Function]}
    isValidValue={[Function]}
    mapping={null}
    onChange={[Function]}
    onError={[Function]}
    onInvalid={[Function]}
    onInvalidSubmit={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
    onValid={[Function]}
    onValidSubmit={[Function]}
    preventExternalInvalidation={false}
    resetValue={[Function]}
    setValidations={[Function]}
    setValue={[Function]}
    showError={[Function]}
    showRequired={[Function]}
    validationErrors={null}
  >
    <WithStyles(Component)
      currentData={
        Object {
          "data": Object {},
          "node": Object {},
        }
      }
      disabled={true}
      formValue={
        Object {
          "data": Object {
            "detail": Object {
              "type": undefined,
            },
            "name": "",
            "type": "Activity",
          },
          "node": Object {
            "customData": Object {
              "start": Object {
                "tempDay": "",
              },
            },
          },
        }
      }
      onChangeFormValue={[Function]}
    />
  </Formsy>
</div>
`;

exports[`<EditEventCard /> renderLeft() should renderLeft 1`] = `
<div>
  <WithStyles(GridItem)>
    <div>
      renderForm
    </div>
  </WithStyles(GridItem)>
</div>
`;

exports[`<EditEventCard /> renderLeft() should renderLeft smDown 1`] = `
<div>
  renderForm
</div>
`;
